rightmesh/test/MeshToken.js

```diff
@@ -268,6 +284,57 @@ contract('MeshToken', (accounts) => {
         });
       });
     });
+
+    /**
+     * Scenario:
+     * 1. Token contract is deployed successfully.
+     * 2. Owner mints the tokens.
+     * 3. Owner paused token transfers.
+     * 4. Owner updates allowedTransfers for an address to true.
+     * 5. Transfer tokens from that address to another should work.
+     */
+    it('should allow transfers when paused for an address that is allowedTransfers', () => {
+      return MeshToken.new().then(meshToken => {
+        return meshToken.mint(accounts[1], 1000).then(() => {
+          return meshToken.pause().then(() => {
+            return meshToken.updateAllowedTransfers(accounts[1], true).then(() => {
+              return meshToken.transfer(accounts[0], 1000, { from: accounts[1] }).then(() => {
+                return meshToken.balanceOf(accounts[0]).then(balance => {
+                  assert.equal(balance, 1000, 'should have the balance of 1000 tokens by now');
+                });
+              });
+            });
+          });
+        });
+      });
+    });
+
+    /**
+     * Scenario:
+     * 1. Token contract is deployed successfully.
+     * 2. Owner mints the tokens.
+     * 3. Owner paused token transfers.
+     * 4. Owner updates allowedTransfers for an address to true.
+     * 4. Owner updates allowedTransfers for the same address to false.
+     * 5. Transfer tokens from that address to another should work.
+     */
+    it('should allow transfers when paused for an address that is allowedTransfers', () => {
```

Comment:
should *not* allow transfers when paused for an address that is *not* allowedTransfers